app/src/main/java/com/example/newsportal/app/topnews/CategoryFragment.kt

```diff
@@ -0,0 +1,47 @@
+package com.example.newsportal.app.topnews
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.RecyclerView
+import com.example.newsportal.utils.NewsAdapter
+import com.example.newsportal.R
+import com.example.newsportal.data.local.model.ArticleLocal
+import org.koin.androidx.viewmodel.ext.android.getViewModel
+
+class CategoryFragment : Fragment() {
+
+    //private val viewModel: NewsViewModel by viewModels({requireParentFragment()})
+    private val viewModel: NewsViewModel by lazy { requireParentFragment().getViewModel<NewsViewModel>() }
+    lateinit var category: String
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_category, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        arguments?.takeIf { it.containsKey("key") }?.apply {

```

Comment:
suggest to made it with by lazy delegate and avoid using lateinit property.
`private val category: String by lazy {
        requireArguments().getString("key", "default value")
    }`

Also consider to define constants for keys, don't hardcode things